shim_maker.py

```diff
@@ -21,62 +26,138 @@ def fancy_list(prefix: str, to_join: Iterable[str], limit: int = 100, with_brack
+            with open(path / mod, "r", encoding="utf-8") as fr:
+                try:
+                    with open(shim_path / mod, "r", encoding="utf-8") as fw:
+                        if fw.read() == fr.read():
+                            continue
+                except FileNotFoundError:
+                    pass
+                fr.seek(0)
+                print("Updating file: ", shim_path / mod)
+                with open(shim_path / mod, "w", encoding="utf-8") as fw:
+                    fw.write(fr.read())
+            modified.append(shim_path / mod)
+
+        elif fn.suffix == ".py":
+            docstring, imports = create_file(f"{pypath}.{mod.stem}")
+            to_write = ""
+            if docstring:
+                to_write += '"""\n' + docstring.strip() + '\n"""\n\n'
+            to_write += imports
+            try:
+                with open(shim_path / mod) as f:
+                    existing = f.read()
+            except FileNotFoundError:
+                existing = None
+            if existing != to_write:
+                print("Updating file: ", shim_path / mod)
+                with open(shim_path / mod, "w", encoding="utf-8") as f:
+                    f.write(to_write)
+
+                modified.append(shim_path / mod)
+
+        elif fn.is_dir():
+            (shim_path / mod).mkdir(parents=True, exist_ok=True)
+            modified.extend(shim_folder(fn, f"{pypath}.{fn.stem}", shim_path / mod))
+
+    return modified
+
+
+def shim(path: pathlib.Path, shim_path: pathlib.Path) -> None:
+    shim_path.mkdir(parents=True, exist_ok=True)
+    return shim_folder(path, path.stem, shim_path)
+
+
+def main():
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("path", help="Path to the disnake module.")
+    parser.add_argument(
+        "-s",
+        "--shim",
+        help="Name of the shim module.",
+        default="discord",
+    )
+    args = parser.parse_args()
+    try:
+        path = pathlib.Path(args.path)
+    except ValueError:
+        print(f"Invalid path to package to be shimmed: {args.path}")
+        return 2
 
-        elif fn.endswith(".py"):
-            imports = create_imports(f"{pypath}.{fn[:-3]}")
-            with open(f"{shim_path}/{fn}", "w", encoding="utf-8") as f:
-                f.write(imports)
+    path = path.resolve()
 
-        elif os.path.isdir(f"{path}/{fn}"):
-            os.makedirs(f"{shim_path}/{fn}", exist_ok=True)
-            shim_folder(f"{path}/{fn}", f"{pypath}.{fn}", f"{shim_path}/{fn}")
+    try:
+        shim_dir = pathlib.Path(args.shim)
+    except ValueError:
+        print(f"Invalid path to new package: {args.shim}")
+        return 2
 
+    shim_dir = shim_dir.resolve()
 
-def shim(path: str, shim_path: str) -> None:
-    os.makedirs("discord", exist_ok=True)
-    shim_folder(path, path.rstrip("/").split("/")[-1], shim_path)
+    edited = shim(path, shim_dir)
+    return edited or None
```

Comment:
```suggestion
    return edited
```

Reply:
This is intentional because I don't want to return an empty list.